lib/seahorn/BvOpSem2.cc

```diff
@@ -789,6 +800,40 @@ class OpSemMemManager {
     return res;
   }
 
+  Expr storeUnalignedIntToMem(Expr val, PtrTy ptr, Expr mem, unsigned byteSz) {
+    unsigned offsetBits = getByteAlignmentBits();
+    assert(offsetBits != 0);
+
+    Expr res;
+    for (unsigned i = 0; i < byteSz; i++) {
+      PtrTy wordAddress = bv::extract(ptrSzInBits() - 1, offsetBits, ptrAdd(ptr, i));
+      PtrTy byteOffset = bv::extract(offsetBits - 1, 0, ptrAdd(ptr, i));
+      byteOffset = bv::zext(byteOffset, wordSzInBits() - 3);
+      // (x << 3) to get bit offset; zero extend to maintain word size
+      PtrTy bitOffset = bv::concat(byteOffset, bv::bvnum(0, 3, ptr->efac()));
+
+      // existing word at address
+      PtrTy alignedPtr = bv::concat(wordAddress, bv::bvnum(0L, offsetBits, ptr->efac()));
+      Expr existingWord = loadAlignedWordFromMem(alignedPtr, mem);
+
+      // set a byte of existing word to 0, byte should be address aligned
```

Comment:
this should be factored out into a separate method since there are multiple ways to implement it